qcodes/instrument_drivers/Keysight/Keysight_N5230A.py

```diff
@@ -0,0 +1,160 @@
+                           vals=Numbers(-np.pi, np.pi),
+                           get_parser=float,
+                           unit='rad')
+
+        self.add_parameter('power',
+                           label='Power',
+                           get_cmd='SOUR:POW?',
+                           set_cmd='SOUR:POW {:.16f}',
+                           docstring='Output power of the VNA output port in dBm.',
+                           get_parser=float,
+                           unit='dBm')
+
+        self.add_parameter('rf_output',
+                           label='RF Output',
+                           docstring='RF Output power on or off.',
+                           get_cmd='OUTP?',
+                           set_cmd='OUTP {}',
+                           val_mapping={'on': 1, 'off': 0})
+
+        self.add_parameter('electrical_delay',
+                           label='Electrical Delay',
+                           get_cmd=':CALC:CORR:EDEL:TIME?',
+                           set_cmd=':CALC:CORR:EDEL:TIME {:.16f}',
+                           get_parser=float,
+                           unit='s')
+
+        self.add_parameter('bandwidth',
+                           label='Bandwidth',
+                           get_cmd=':SENS:BAND?',
+                           set_cmd=':SENS:BAND {:.16f}',
+                           get_parser=float,
+                           unit='Hz')
+
+        self.add_parameter('center_frequency',
+                           label='Center Frequency',
+                           get_cmd=':SENS:FREQ:CENT?',
+                           set_cmd=':SENS:FREQ:CENT {}',
+                           get_parser=float,
+                           unit='Hz')
+
+        self.add_parameter('start_frequency',
+                           label='Start Frequency',
+                           get_cmd=':SENS:FREQ:STAR?',
+                           set_cmd=':SENS:FREQ:STAR {}',
+                           get_parser=float,
+                           unit='Hz')
+
+        self.add_parameter('stop_frequency',
+                           label='Stop Frequency',
+                           get_cmd=':SENS:FREQ:STOP?',
+                           set_cmd=':SENS:FREQ:STOP {}',
+                           get_parser=float,
+                           unit='Hz')
+
+        self.add_parameter('span',
+                           label='Span',
+                           get_cmd=':SENS:FREQ:SPAN?',
+                           set_cmd=':SENS:FREQ:SPAN {}',
+                           get_parser=float,
+                           unit='Hz')
+
+        self.add_parameter('s_parameter',
+                           label='S parameter',
+                           docstring='Set the measured S parameter, allowed values are "S11", "S12", "S21", and "S22"',
+                           get_cmd=':CALC:PAR:CAT?',
+                           set_cmd=':CALC:PAR:MOD  {}',
+                           get_parser=lambda x: x[1:-1].split(',')[1])
```

Comment:
should there be an explanation for this parser so that other developers know the reasoning behind it?